feat: autoconnect try out

toniocodo opened the pull request:
❗DO NOT MERGE: this is only to get a test deploy

dimsome commented:
Couple of small things:

- When wallet is connected and switched network to an unsupported network, the wallet disconnects completely (happens in certain conditions)
- Should instead show that it is an unsupported network

- When wrong network is displayed this modal is shown on click. Button should be left-aligned or full width
- Additionally, polygon is also supported but not suggested in the main app.
![Screen Shot 2022-06-24 at 22 14 11](https://user-images.githubusercontent.com/69167058/175565148-e768743d-8950-4ab4-b0a7-e019182f977d.png)

rafaelugolini commented on libs/base/src/lib/hooks/useResolveENSName.ts:
```diff
@@ -20,32 +20,38 @@ export const useResolveENSName = (ensName: string): FetchState<string> => {
     provider
       .resolveName(ensName)
-      .then(address => {
+      .then((address: any) => {
```
👀

rafaelugolini commented:
🐛 I noticed that connecting it via gnosis safe is getting my wallet instead of the gnosis one

@toniocodo you can try like this:

https://gnosis-safe.io/app/eth:0xF6FF1F7FCEB2cE6d26687EaaB5988b445d0b94a2/apps?appUrl=https://mstable.app

https://gnosis-safe.io/app/eth:0xF6FF1F7FCEB2cE6d26687EaaB5988b445d0b94a2/apps?appUrl=https://apps-protocol-af0f6--pr183-feat-coinbase-wallet-24xwnhpl.web.app/

rafaelugolini commented:
After claiming the rewards from save, the spinner kept running:
<img width="669" alt="image" src="https://user-images.githubusercontent.com/2651141/175808598-31b66841-3361-46a0-a347-d84294e9d541.png">

toniocodo commented:
> Couple of small things:
> 
> * When wallet is connected and switched network to an unsupported network, the wallet disconnects completely (happens in certain conditions)
> * Should instead show that it is an unsupported network
> * When wrong network is displayed this modal is shown on click. Button should be left-aligned or full width
> * Additionally, polygon is also supported but not suggested in the main app.

I added a network watcher to detect when network is switched and unsupported. The problem is that wagmi, when it detects an unsupported network disconnects you and uses the first available chain.
![image](https://user-images.githubusercontent.com/13216617/175905094-7aa2e898-5d89-4d2d-bbc6-116580da7722.png)
It happens before triggering the network switch (not on all cases though, it appears that it sometimes happen in the correct order), so I cannot avoid the disconnect. I wonder if their latest version fixes that, it may be the case since they reworked the connect hook. I'd rather do the upgrade in another PR though.

I only propose to "reset to eth.mainnet"  or to use wallet to switch network as the supported chains are hard to retrieve when you are disconnected.

Also updated wording and modal accordingly
![image](https://user-images.githubusercontent.com/13216617/175913663-34c4b5e7-1fea-4d57-a24c-155132c1dda6.png)

dimsome commented:
One reported issue: 

_I often have the issue when going to mStable.app when on Polygon, and Polygon is pre-selected already from a previous visit, that mStable App in Metamask prompts me to switch to Mainnet (which is sorta annoying)_

Could we detect if the network in MM is selected to Polygon, that the wallet is not prompted to change to mainnet?

dimsome reviewed (approved):
not much to add from me

toniocodo commented on apps/governance/.gitignore:
```diff
@@ -62,8 +62,5 @@ node_modules/
 .yarn-integrity
 
-# dotenv environment variables file
-.env
```
those are committed .env for application-scope variables, if someone wants to add local env variables, he can
- put them at repo root-level in `.env.local` (as per the README recommend)
- put them in a  application-scoped `.env.local` file that remains ignored (same logic than at the root)

So I think we're good 😃
